Google.Api.Generator.Tests/ProtoTests/RoutingHeadersExplicit/RoutingHeadersExplicit.proto

```diff
@@ -0,0 +1,121 @@
+﻿syntax = "proto3";
+
+package testing.routingheadersexplicit;
+
+option csharp_namespace = "Testing.RoutingHeadersExplicit";
+
+import "google/api/annotations.proto";
+import "google/api/client.proto";
+import "google/api/routing.proto";
+
+service RoutingHeadersExplicit {
+	option (google.api.default_host) = "routingheaders.example.com";
+
+	rpc NoUrlMethod(SimpleRequest) returns(Response);
+
+	rpc PlainNoTemplate(SimpleRequest) returns(Response) {
+		option (google.api.http) = {
+			get: "/v2/{name=projects/*/instances/*/tables/*}:sampleRowKeys"
+		};
+		option (google.api.routing) = {
+			routing_parameters {
+				field: "name"
+			}
+		};
+	}
+
+	rpc PlainFullField(SimpleRequest) returns(Response) {
+		option (google.api.http) = {
+			get: "/v2/{name=projects/*/instances/*/tables/*}:sampleRowKeys"
+		};
+		option (google.api.routing) = {
+			routing_parameters {
+				field: "name"
+				path_template: "{table_name=projects/*/instances/*/tables/*}"
+			}
+		};
+	}
+
+	rpc PlainExtract(SimpleRequest) returns(Response) {
+		option (google.api.http) = {
+			get: "/v2/{name=projects/*/instances/*/tables/*}:sampleRowKeys"
+		};
+		option (google.api.routing) = {
+			routing_parameters {
+				field: "name"
+				path_template: "projects/*/instances/*/{table_name=tables/*}"
+			}
+		};
+	}
+
+	rpc Nested(NestedRequest) returns(Response) {
+		option (google.api.http) = {
+			get: "/v2/{nest1.nest2.name=projects/*/instances/*/tables/*}:withSubMessage"
+		};
+		option (google.api.routing) = {
+			routing_parameters {
+				field: "nest1.nest2.name"
+				path_template: "{table_name=projects/*/instances/*/tables/*}"
+			}
+			routing_parameters {
```

Comment:
Is there a reason why this test has two parameters? (I'd stick to just one to test nesting, and use the complex one for multiple parameters, unless there's something I'm missing.)

Reply:
It served a purpose for the previous design, now it is refactored to ... still have two parameters but to showcase a different thing entirely.